internal/pkg/utils/keybase.go

```diff
@@ -0,0 +1,143 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/cisco-sso/kdk/internal/pkg/utils/simpleprompt"
+	"github.com/codeskyblue/go-sh"
+)
+
+// Write keybase mirror script [windows only]
+func keybaseWriteMirrorScript(configDir string) (out string, err error) {
+
+	cmdString := `
+@echo off
+
+if "%1"=="" (
+  echo "You must pass either start or stop"
+  break
+)
+
+if "%1"=="start" (
+  echo "Starting"
+  start "KDK Keybase Mirror" /B "C:\Program Files\Dokan\Dokan Library-1.1.0\sample\mirror\mirror.exe" /r K:\ /l C:\Users\%USERNAME%\.kdk\keybase
+  break
+)
+
+if "%1"=="stop" (
+  echo "stopping"
+  tskill.exe mirror
+  break
+) else (
+  echo "Unrecognized parameter %1.  You must pass either start or stop"
+)
+`
+	script := []byte(cmdString)
+
+	scriptPath := filepath.Join(configDir, "keybase-mirror.cmd")
+	if _, err := os.Stat(scriptPath); os.IsNotExist(err) {
+		err := ioutil.WriteFile(scriptPath, script, 0700)
+		if err != nil {
+			return "", err
+		}
+	}
+	return scriptPath, nil
+}
+
+// Start keybase mirror [windows only]
+func KeybaseStartMirror(configDir string) error {
+
+	keybaseTestDir := filepath.Join(configDir, "keybase", "private")
+	if _, err := os.Stat(keybaseTestDir); err == nil {
+		return nil
+	}
+	scriptPath, err := keybaseWriteMirrorScript(configDir)
+	if err != nil {
+		return err
+	}
+	commandString := fmt.Sprintf("powershell %s %s", scriptPath, "start")
+	commandMap := strings.Split(commandString, " ")
+	if err := sh.Command(commandMap[0], commandMap[1:]).SetStdin(os.Stdin).Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Stop keybase mirror [windows only]
+func KeybaseStopMirror(configDir string) error {
+
+	keybaseTestDir := filepath.Join(configDir, "keybase", "private")
+	if _, err := os.Stat(keybaseTestDir); err != nil {
```

Comment:
MUST: This check isn't valid, because keybase may have been stopped (k: disappeared), but the mirror could still be running.  You would be breaking earlier.  Instead, you should check if the mirror is running and just kill it. 

Please confirm how this works with multiple kdk's.

Reply:
We can't use StopMirror with multiple KDKs.  See comment above